File: CSharp-MCP-Server/WeatherTools.cs
using MCPSharp;

namespace WeatherMcp;

[McpTool("weather_tools", "Tools for getting weather information")]
public class WeatherTools
{
    [McpFunction("getWeather", "Gets the weather for a given zipcode")]
    public static Weather getWeather(string zipcode)
    {
        return new Weather("Cloudy and drizzly", "72.123F");
    }
}

File: ruby/ruby_llm/tools/WeatherTools.rb
require 'ruby_llm'

Weather = Data.define(:conditions, :temperature)

class WeatherTools < RubyLLM::Tool
    description "Gets the weather for a given zipcode"

    param :zipcode,
        type: :string,
        desc: "The zipcode to get the weather for"

    def execute(zipcode:)
        Weather.new("Partly cloudy", "71F")
    end
end